core/lib/vm_interface/Cargo.toml

```diff
@@ -14,6 +14,8 @@ categories.workspace = true
 zksync_contracts.workspace = true
 zksync_system_constants.workspace = true
 zksync_types.workspace = true
+zksync_utils.workspace = true
+zksync_mini_merkle_tree.workspace = true
```

Comment:
If possible, I'd advise against adding either of these 2 deps, since it makes the interface crate too heavy. Could it be possible e.g. to add `PubdataBuilder` implementations to the executor crate?

Reply:
I've moved impls to `multivm`, otherwise it's not possible to run `vm.finish_batch` in multivm tests. It also required moving vm dump playback functions.